nautobot/extras/models/mixins.py

```diff
@@ -18,3 +25,51 @@ def dynamic_groups(self):
             self._dynamic_group_queryset = queryset
 
         return self._dynamic_group_queryset
+
+
+# 2.0 TODO: Remove after v2 since we will no longer care about backwards-incompatibility.
+class TaggableManagerMonkeyMixin:
+    """
+    Dynamically-applied monkey-patch mixin that is used to replace any defined
+    methods on eligible objects.
+
+    Intended to be used on the right-hand side of `monkey_mix()` to overload the
+    `formfield()` method.
+
+    Usage:
+
+        monkey_mix(TaggableManager, TaggableManagerMonkeyMixin)
+
+    See: `nautobot.extras.apps.ready()`
+    """
+
+    @once_per("cls")
+    def _install_hotfix(self, cls):
+        # Install auto-created models as their module attributes to make them picklable
+        module = sys.modules[cls.__module__]
+        if not hasattr(module, cls.__name__):
+            setattr(module, cls.__name__, cls)
+
+    # This is probably still needed if models are created dynamically.
+    def contribute_to_class(self, cls, name):
+        self._no_monkey.contribute_to_class(self, cls, name)
+        # Django migrations create lots of fake models, just skip them
+        # NOTE: we make it here rather then inside _install_hotfix()
+        #       because we don't want @once_per() to hold refs to all of them.
+        if cls.__module__ != "__fake__" and family_has_profile(cls):
```

Comment:
What's the purpose/significance of `family_has_profile`? Can we get some comments documenting that?

Reply:
I swear I replied to this before but I'm not seeing my response! 

This code was fork-lifted as is from [`cacheops.query.ManagerMixin`](https://github.com/Suor/django-cacheops/blob/master/cacheops/query.py#L409-L429), including the use of [`family_has_profile`](https://github.com/Suor/django-cacheops/blob/ad83e51b6d82ba2bf4820953925ff889e4c4b840/cacheops/utils.py#L31) which underneath is used to identify ["all proxy models, including subclasess, superclassses and siblings"](https://github.com/Suor/django-cacheops/blob/ad83e51b6d82ba2bf4820953925ff889e4c4b840/cacheops/utils.py#L16-L27) for a model object. It's how it's deciding whether to patch an object or not.

Reply:
Can we get a comment added to this effect?